Source/ACE.Database/IShardDatabase.cs

```diff
@@ -34,5 +34,8 @@ public interface IShardDatabase : ICommonDatabase
         uint GetNextCharacterId();
 
         AceCharacter GetCharacter(uint id);
+
+        // call back error interfaces
+        void UnableToSaveObject(AceObject ao);
```

Comment:
This isn't part of the database interface - you need a callback to the calling code, not just some other function in your current class.